x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/EnterpriseSearch.java

```diff
@@ -85,7 +111,12 @@ public Collection<Object> createComponents(
         if (enabled == false) {
             return Collections.emptyList();
         }
-        final EngineIndexService engineService = new EngineIndexService(client, clusterService, namedWriteableRegistry, null);
+        final EngineIndexService engineService = new EngineIndexService(
+            client,
+            clusterService,
+            namedWriteableRegistry,
+            BigArrays.NON_RECYCLING_INSTANCE
```

Comment:
I couldn't find any other way of setting a BigArrays instance, is this OK?

Reply:
This looks consistent with how its done elsewhere in the code base.

Reply:
It's not ok for the finished product, we should use the real `BigArrays` which recycles pages. But it is ok for now.

Reply:
> should we keep it as a Component, to avoid having to explicitly create it in the TransportActions?

It's just a client wrapper so I think it's ok to create it on each TransportAction for now.

Reply:
Can you add a TODO here so that we don't forget?